Lesson 1

aphmelev opened the pull request.

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
start

DmitryChitalov commented on Урок 1. Практическое задание/task_1.py:
```diff
@@ -22,10 +22,10 @@ def check_1(lst_obj):
-    return lst_to_set  # !!!
+    lst_to_set = set(lst_obj)  # O(n)
+    return lst_to_set  # O(1)
 
```
выполнено

DmitryChitalov commented on Урок 1. Практическое задание/task_1.py:
```diff
@@ -37,12 +37,12 @@ def check_2(lst_obj):
+        if lst_obj[j] in lst_obj[j+1:]:    # O(n)
+            return False                   # O(1)
+    return True                            # O(1)
 
```
выполнено

DmitryChitalov commented on Урок 1. Практическое задание/task_1.py:
```diff
@@ -53,14 +53,16 @@ def check_3(lst_obj):
+    lst_copy = list(lst_obj)                 # O(n)
+    lst_copy.sort()                          # O( n log n)
+    for i in range(len(lst_obj) - 1):        # O(n)
+        if lst_copy[i] == lst_copy[i+1]:     # O(n)
```
if lst_copy[i] == lst_copy[i+1]:     # O(n)
неверно

DmitryChitalov commented on Урок 1. Практическое задание/task_2.py:
```diff
@@ -4,12 +4,41 @@
+        if i < first_element:
+            first_element = i
+    return first_element
+
```
неверно
здесь тоже будет линейная
вот проставьте сложности каждого выражения

DmitryChitalov commented on Урок 1. Практическое задание/task_3.py:
```diff
@@ -17,3 +17,35 @@
+    top3_company = {}                       # O(1)
+    dict_list = list(dict.values())         # O(n)
+    dict_list.sort(reverse=True)            # O( n log n)
+    for el in dict_list[:3]:                # O(n)
```
for el in dict_list[:3]:                # O(n)
константная

DmitryChitalov commented on Урок 1. Практическое задание/task_3.py:
```diff
@@ -17,3 +17,35 @@
+    dict_list.sort(reverse=True)            # O( n log n)
+    for el in dict_list[:3]:                # O(n)
+        for name, prof in dict.items():     # O(n)
+            if prof == el:                  # O(n)
```
if prof == el:                  # O(n)
константная

DmitryChitalov commented on Урок 1. Практическое задание/task_3.py:
```diff
@@ -17,3 +17,35 @@
+    'ООО "Сибиряки': 17300}
+
+
+def top3_company_v1(dict):                  # O( n log n) или O(n**4) так и не понял как суммировать
```
( n log n) - итоговая

DmitryChitalov commented on Урок 1. Практическое задание/task_4.py:
```diff
@@ -20,3 +20,51 @@
+def log_in_v1(users_list=users):  # O(n)
+    username = input('Your login: ')  # O(1)
+    password = input('Your password: ')  # O(1)
+    if users_list.get(username):  # O(n)
```
if users_list.get(username):  # O(n)
неверно
не посмотрели таблицу сложностей

DmitryChitalov commented on Урок 1. Практическое задание/task_4.py:
```diff
@@ -20,3 +20,51 @@
+}
+
+
+def log_in_v1(users_list=users):  # O(n)
```
какая итоговая будет здесь?

DmitryChitalov commented on Урок 1. Практическое задание/task_5.py:
```diff
@@ -18,3 +18,33 @@
+
+    def print_stack(self):
+        print(self.list)
+
```
это не стек
не все необходимые методы реализованы

DmitryChitalov commented on Урок 1. Практическое задание/task_6.py:
```diff
@@ -15,3 +15,79 @@
+print(task_to_done_3)
+board.done_task(task_to_done_3)
+
+print(board.size())
```
выполнено

DmitryChitalov commented on Урок 1. Практическое задание/task_7.py:
```diff
@@ -14,3 +14,24 @@
+    return still_equal
+
+
+print(pal_checker("молоко делили ледоколом"))
```
выполнено

Commit:
изменение в задаче 1 алгоритм 3: if lst_copy[i] == lst_copy[i+1]:     # O(1)

Commit:
изменение в задаче 2 алгоритм 1: изменил на 2 цикла, теперь квадратичная

Commit:
изменение в задаче 3 алгоритм top3_company_v1: изменил сложности, но не понимаю почему for el in dict_list[:3]: - константаня

Commit:
изменение в задаче 4
def log_in_v1(users_list=users): - сложность изменена на квадратичную
if users_list.get(username): изменена на константаня

Commit:
изменение в задаче 5
добавлены методы удаления стопки, просмотр последнего елемента (количества тарелок) в последней стопке и вывод информации о текущем состоянии стопок и стеков

DmitryChitalov commented on Урок 1. Практическое задание/task_2.py:
```diff
@@ -4,12 +4,43 @@
+
+"""
 Примечание: ПРОШУ ВАС ВНИМАТЕЛЬНО ЧИТАТЬ ЗАДАНИЕ!
 -- нельзя использовать встроенные функции min() и sort()
```
list-встроенное имя
нарушение пеп-8

DmitryChitalov commented on Урок 1. Практическое задание/task_3.py:
```diff
@@ -17,3 +17,35 @@
+    top3_company = {}                       # O(1)
+    dict_list = list(dict.values())         # O(n)
+    dict_list.sort(reverse=True)            # O( n log n)
+    for el in dict_list[:3]:                # O(1) вы комментировали ДЗ "констанстная сложность", не понимаю почему Т_Т
```
for el in dict_list[:3]:                # O(1) вы комментировали ДЗ "констанстная сложность", не понимаю почему Т_Т
потому что сколько будет итераций в цикле?
3. и всегда будет 3
мы это знаем заранее

DmitryChitalov commented on Урок 1. Практическое задание/task_3.py:
```diff
@@ -17,3 +17,35 @@
+    return top3_company                     # O(1)
+
+
+def top3_company_v2(dict):                                      # O(n)
```
dict
встроенное имя

DmitryChitalov commented on Урок 1. Практическое задание/task_4.py:
```diff
@@ -20,3 +20,51 @@
+}
+
+
+def log_in_v1(users_list=users):  # O(n^2)
```
def log_in_v1(users_list=users):  # O(n^2)
итоговая не квадратичная

DmitryChitalov commented on Урок 1. Практическое задание/task_4.py:
```diff
@@ -20,3 +20,51 @@
+    username = input('Your login: ')  # O(1)
+    password = input('Your password: ')  # O(1)
+    if users_list.get(username):  # O(1)
+        if users_list[username]['Password'] == password and users_list[username]['Activate']:  # O(n)
```
if users_list[username]['Password'] == password and users_list[username]['Activate']:  # O(n)
здесь константная

DmitryChitalov commented on Урок 1. Практическое задание/task_4.py:
```diff
@@ -20,3 +20,51 @@
+    if users_list.get(username):  # O(1)
+        if users_list[username]['Password'] == password and users_list[username]['Activate']:  # O(n)
+            return print('Вход осуществлен')  # O(1)
+        elif users_list[username]['Password'] == password and not users_list[username]['Activate']:  # O(n)
```
elif users_list[username]['Password'] == password and not users_list[username]['Activate']:  # O(n)
констмантная

DmitryChitalov commented on Урок 1. Практическое задание/task_4.py:
```diff
@@ -20,3 +20,51 @@
+log_in_v2('Andrey', '123456')
+log_in_v2('Daniil', '123456')
+log_in_v2('Andrey', '124356')
+log_in_v2('Kiril', '654321')
```
здесь верно

DmitryChitalov commented on Урок 1. Практическое задание/task_5.py:
```diff
@@ -18,3 +18,58 @@
+
+class Stack:
+    def __init__(self):
+        self.list = []
```
list 
встроенное имя

Commit:
Задача 1

Commit:
Задача 2

Commit:
Задача 3

Commit:
Задача 4

Commit:
Задача 5

Commit:
Задача 5

Commit:
Задача 6

Commit:
Задача 6 доработка проверки на число

Commit:
Задача 6 доработка проверки на число

Commit:
Задача 7

Commit:
Задача 4 доработка на проверку числа
